### PR TITLE
Add prettier to the codebase and format code

### LINKED ISSUE #95: Add Prettier
Add prettier code formatter, being careful to not cause compatibility issues with eslint/tslint.

### PR DESCRIPTION
**Pull request by JackWilb**
Closes #95.

This PR makes `yarn lint` use the `--no-fix` option by default, adds prettier, adds the commands to run prettier (`format:check`, `format:fix`), formats the code, and adds the format check to the CI.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "overrides": [
+        {
+            "files": ["**/*.css", "**/*.scss", "**/*.html"],
+            "options": {
+                "singleQuote": false
+            }
+        }
+    ],
+    "trailingComma": "all"
+}
```
Comment (waxlamp) [inline .prettierrc:12]: Could you add a newline to the end of this file?

**Review by waxlamp — CHANGES_REQUESTED** [review]
Love it! Thanks for getting this together, Jack.

One super minor nitpick, perfect otherwise.

**Comment by waxlamp** [comment]
attn @gotdairyya, @mozartfish: once this is merged, we can rebase your in-progress work on this change to get everything synced up. Let me know if you need/want help with that.

File: .prettierrc (hunk 1 above)
Comment (JackWilb) [reply .prettierrc]: See 949d4e4

**Review by waxlamp — APPROVED** [review]

**Pull request merged**